specs/skeleton.md

```diff
@@ -0,0 +1,99 @@
+# Skeleton
+
+## Overview
+
+A skeleton is used as a visual placeholder for an element while it is in a loading state and usually presents itself as a simplified wireframe-like version of the UI it is representing.
+
+### Background
+
+- [Github Issue #3755](https://github.com/microsoft/fast/issues/3755)
+
+### Use Cases
+
+Used to give users a sense of loading progress activity rather than merely presenting a blank page or container.
+
+### Non-goals
+
+#### Specfic Skeletons
+
+Should not be made to reflect a specific element (e.g. `skeleton-button`, `skeleton-text`, `skeleton-heading`, etc...). The skeleton should be as generic of a container as possible as to represent various elements of different shapes and sizes.
+  
+### Features
+
+- `Shape`: Circle or rect shapes can be selected. Border radius of rectagled shape will be determined by the design system.
+
+### Prior Art/Examples
+
+- [Ant Design](https://ant.design/components/skeleton/)
+- [Carbon](https://www.carbondesignsystem.com/patterns/loading-pattern#skeleton-states)
+- [FluentUI (Shimmer)](https://developer.microsoft.com/en-us/fluentui#/controls/web/shimmer)
+- [Material UI](https://material-ui.com/components/skeleton/)
+- [Kalo](https://kalo.design/components/skeleton/)
+- [Pajamas](https://design.gitlab.com/components/skeleton-loader)
+- [Polaris](https://polaris.shopify.com/components/feedback-indicators/skeleton-thumbnail)
+- [React Loading Skeleton](https://github.com/dvtng/react-loading-skeleton#readme)
+- [Semantic UI (Placeholder)](https://semantic-ui.com/elements/placeholder.html)
+- [Telus](https://tds.telus.com/community/index.html#skeleton)
+- [Thumbprint](https://thumbprint.design/guide/product/loaders/#example-2.-skeleton-loader)
+- [Vercel](https://vercel.com/design/skeleton)
+
+### API
+
+#### Component Name
+- `fast-avatar`
+
+#### Attributes
+|   Name    | Description                                                 | Type                                |
+|-----------|-------------------------------------------------------------|-------------------------------------|
+| `shape`   | Determines the avatar coin shape. Rectangle by default.     | `string: rect (default) | circle`   |
+| `fill`    | Sets the background fill of the avatar coin.                | `string: hex color`                 |
```

Comment:
It seems like the fill would be a complicated value, presumably a gradient and presumably based on the design system so the color coordinates with the rest of the app design system.

Reply:
Valid. I might have to explore this more.